zebra-state/src/service.rs

```diff
@@ -193,6 +197,61 @@ impl StateService {
         // TODO: contextual validation design and implementation
         Ok(())
     }
+
+    /// Create a block locator for the current best chain.
+    fn block_locator(&self) -> Option<Vec<block::Hash>> {
+        let tip_height = self.tip()?.0;
+
+        let heights = crate::util::block_locator_heights(tip_height);
```

Comment:
We could probably fold the `block_locator_heights` method into this one, and delete the entire `util` module.